src/mapgen.cpp

```diff
@@ -1707,14 +1707,19 @@ class jmapgen_field : public jmapgen_piece
 {
     public:
         mapgen_value<field_type_id> ftype;
-        int intensity;
+        std::vector<int> intensities;
         time_duration age;
         bool remove;
         jmapgen_field( const JsonObject &jsi, const std::string &/*context*/ ) :
             ftype( jsi.get_member( "field" ) )
-            , intensity( jsi.get_int( "intensity", 1 ) )
             , age( time_duration::from_turns( jsi.get_int( "age", 0 ) ) )
             , remove( jsi.get_bool( "remove", false ) ) {
+            for( JsonValue jv : jsi.get_array( "intensity" ) ) {
+                intensities.push_back( jv.get_int() );
+            }
```

Comment:
```suggestion
            if ( jsi.has_array( "intensity" ) ) {
                for( JsonValue jv : jsi.get_array( "intensity" ) ) {
                    intensities.push_back( jv.get_int() );
                }
            } 
```

I think doing this should be enough. This checks if the given value is an array before trying to read it. And if it's not the fallback below will be triggered to read an int.